File: HubWebApp/Internal/XTI_Hub/AppLogEntry.cs
using XTI_App.Abstractions;
using XTI_Core;
using XTI_HubDB.Entities;

namespace XTI_Hub;

public sealed class AppLogEntry
{
    private readonly AppLogEntryEntity record;

    internal AppLogEntry(AppLogEntryEntity record)
    {
        this.record = record ?? new AppLogEntryEntity();
        ID = this.record.ID;
    }

    public int ID { get; }
    public string Caption { get => record.Caption; }
    public string Message { get => record.Message; }
    public string Detail { get => record.Detail; }
    public AppEventSeverity Severity() => AppEventSeverity.Values.Value(record.Severity);

    public AppLogEntryModel ToModel() => new AppLogEntryModel
    {
        ID = ID,
        RequestID = record.RequestID,
        TimeOccurred = record.TimeOccurred,
        Severity = Severity(),
        Caption = Caption,
        Message = Message,
        Detail = Detail
    };

    public override string ToString() => $"{nameof(AppLogEntry)} {ID}";
}

File: HubWebApp/Apps/HubWebApp/Scripts/Lib/AppLogEntry.ts
import { DateTimeOffset } from "@jasonbenfield/sharedwebapp/DateTimeOffset";
import { AppEventSeverity } from "./Http/AppEventSeverity";

export class AppLogEntry {
    readonly id: number;
    readonly requestID: number;
    readonly timeOccurred: DateTimeOffset;
    readonly severity: AppEventSeverity;
    readonly caption: string;
    readonly message: string;
    readonly detail: string;
    readonly category: string;
    readonly actualCount: number;

    constructor(readonly source?: IAppLogEntryModel) {
        this.id = source ? source.ID : 0;
        this.requestID = source ? source.RequestID : 0;
        this.timeOccurred = source ? source.TimeOccurred : DateTimeOffset.max();
        this.severity = source ? AppEventSeverity.values.value(source.Severity) : AppEventSeverity.values.NotSet;
        this.caption = source ? source.Caption : "";
        this.message = source ? source.Message : "";
        this.detail = source ? source.Detail : "";
        this.category = source ? source.Category : "";
        this.actualCount = source ? source.ActualCount : 0;
    }
}